core/src/display_object.rs

```diff
@@ -1391,6 +1391,19 @@ pub trait TDisplayObject<'gc>:
                 }
             }
         }
+
+        if let Some(movie) = self.as_movie_clip() {
+            if let Some(obj) = movie.object2().as_object() {
+                let mut activation = Avm2Activation::from_nothing(context.reborrow());
+                let movieclip_class = activation.avm2().classes().movieclip;
```

Comment:
could this be instead done by setting some flag at instantiation time? 

(I'm mostly not a fan of making a new reborrow on possibly-each MC construct phase; one other way to fix it could be to somehow remove `activation` requirement from `is_of_type`; that said, setting some MC flag still sounds like a "nicer" solution in general.)

Reply:
I changed it to take `UpdateContext` in https://github.com/ruffle-rs/ruffle/pull/10735